### PR TITLE
CAPT-1702 Migrate test environment to AKS

### PR DESCRIPTION
**Pull request by AbigailMcP**
## Changes in this PR

Create test environment, so we can have proper testing before deploying to production

## Guidance to review

Run `make test-aks terraform-plan-aks` and check output is correct

Testing URLS (created by running make commands locally):

https://claim-additional-payments-for-teaching-test-web.test.teacherservices.cloud/additional-payments/claim
https://claim-additional-payments-for-teaching-test-web.test.teacherservices.cloud/student-loads/claim
https://claim-additional-payments-for-teaching-test-web.test.teacherservices.cloud/admin

### CODE REVIEW

**Comment by slorek** [comment]
Had a look at the currently deployed test environment:

* We need to add claim-additional-payments-for-teaching-test-web.test.teacherservices.cloud to the DfE Sign In white list on their test environment (unable to log in to admin)
* I'm getting 404 errors on /additional-payments/landing-page and /additional-payments/claim

**Comment by AbigailMcP** [comment]
**Tested:**
✅ Basic auth
✅ Claim submission
✅ Exception reporting to Rollbar
✅ Ordnance survey API address lookup
✅ HMRC API integration
✅ Notify emails sent
✅ Checked application and deployment logs for errors
✅ DQT integration (tested in rails console)

**Issues:**
❌ DfE Identity sign in
_the specified 'redirect_uri' is not valid for this client application_
whitelisting requested

❌ Admin login with DfE Sign In
_redirect_uri did not match any of the client's registered redirect_uris_

**Untested (need admin login):**
🟠 Levelling Up Premium Payments awards upload
🟠 TPS data upload
🟠 School data import

**Other things needed:**
- [ ] Testing of the deploy workflow
- [ ] BigQuery integration - could be follow up PR
- [ ] StatusCake integration - could be follow up PR
- [ ] Update documentation (accessing cluster, running rake tasks etc) - could be follow up PR

**Review by slorek — APPROVED** [review]

File: .github/workflows/build_and_deploy.yml [hunk 1]
```diff
@@ -78,3 +80,36 @@ jobs:
...
+        with:
+          environment: test-aks
+          docker-image: ${{ needs.build.outputs.docker-image-tag }}
+          azure-credentials: ${{ secrets.AZURE_CREDENTIALS }}
+
+      - name: Run migrations
+        shell: bash
+        run: |
+          make ci test-aks get-cluster-credentials
+          kubectl exec -n srtl-test deployment/claim-additional-payments-for-teaching-test-worker -- sh -c "DISABLE_DATABASE_ENVIRONMENT_CHECK=1 bin/prepare-database"
```
Comment (RMcVelia) [inline .github/workflows/build_and_deploy.yml:115]: We normally have a smoke test post deployment for permanent env deployments (i.e. test/prod), and also look to have a slack msg generated on workflow failure (of perm env deploy/smoke test failure)

File: terraform/application/variables.tf [hunk 2]
```diff
@@ -60,6 +60,10 @@ variable "canonical_hostname" {
   description = "External domain name for the app service"
   default     = null
 }
+variable "app_replicas" {
+  description = "Number of replicas of the web app"
+  default = 1
+}
```
[the remark this answers is not in the document]
Comment (AbigailMcP) [reply terraform/application/variables.tf:66]: that sounds like a good way to go! i'll update :)

[the remark this answers is not in the document]
Comment (AbigailMcP) [reply terraform/application/variables.tf]: https://github.com/DFE-Digital/claim-additional-payments-for-teaching/pull/2815/commits/73bff5b1bc396d2560611b2f7be6e2e225aed660

File: .github/workflows/build_and_deploy.yml (hunk 1 above)
Comment (AbigailMcP) [reply .github/workflows/build_and_deploy.yml:115]: https://github.com/DFE-Digital/claim-additional-payments-for-teaching/pull/2875

Comment (AbigailMcP) [reply .github/workflows/build_and_deploy.yml:115]: slack message PR: https://github.com/DFE-Digital/claim-additional-payments-for-teaching/pull/2864

**Pull request merged**